arangodb-net-standard/Transport/Http/HttpApiTransport.cs

```diff
@@ -33,26 +33,90 @@ public HttpApiTransport(HttpClient client)
+            Uri hostUri,
+            string dbName)
+        {
+            var client = new HttpClient();
+            client.BaseAddress = new Uri(hostUri.AbsoluteUri + "/_db/" + dbName + "/");
+
+            var transport = new HttpApiTransport(client);
+            return transport;
+        }
+
+        /// <summary>
+        /// Get an instance of <see cref="HttpApiTransport"/> that uses basic auth.
         /// </summary>
         /// <param name="hostUri"></param>
         /// <param name="dbName"></param>
         /// <param name="username"></param>
         /// <param name="passwd"></param>
-        public HttpApiTransport(
+        public static HttpApiTransport UsingBasicAuth(
             Uri hostUri,
             string dbName,
             string username,
             string passwd)
         {
             var client = new HttpClient();
             client.BaseAddress = new Uri(hostUri.AbsoluteUri + "/_db/" + dbName + "/");
-            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+
+            var transport = new HttpApiTransport(client);
+            transport.SetBasicAuth(username, passwd);
+
+            return transport;
+        }
+
+        /// <summary>
+        /// Get an instance of <see cref="HttpApiTransport"/> that uses
+        /// JWT-Token authentication.
+        /// </summary>
+        /// <param name="hostUri"></param>
+        /// <param name="dbName"></param>
+        /// <param name="jwtToken"></param>
+        /// <returns></returns>
+        public static HttpApiTransport UsingJwtAuth(
+            Uri hostUri,
+            string dbName,
+            string jwtToken)
+        {
+            var client = new HttpClient();
+            client.BaseAddress = new Uri(hostUri.AbsoluteUri + "_db/" + dbName + "/");
+
+            var transport = new HttpApiTransport(client);
+            transport.SetJwtToken(jwtToken);
+
+            return transport;
+        }
+
+        /// <summary>
+        /// When using Basic auth, call this method to set the username and password
+        /// used in requests to ArangoDB.
+        /// </summary>
+        /// <param name="username"></param>
+        /// <param name="passwd"></param>
+        public void SetBasicAuth(string username, string passwd)
+        {
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
                 "Basic",
                 Convert.ToBase64String(
                     Encoding.ASCII.GetBytes($"{username}:{passwd}")));
+        }
 
-            _client = client;
+        /// <summary>
+        /// When using JWT authentication, call this method to refresh the JWT token
+        /// used in requests to ArangoDB.
+        /// </summary>
+        /// <param name="jwt"></param>
+        public void SetJwtToken(string jwt)
+        {
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                   "bearer",
```

Comment:
Should we use `Bearer` here? Not that the lower case version may cause any problem but I've always seen the uppercase version in the RFCs.

Reply:
This is as documented by ArangoDB:  https://www.arangodb.com/docs/stable/http/general.html#authentication-via-jwt

Reply:
Ok, I wish we would follow the RFC though but no big deal.

https://tools.ietf.org/html/rfc6750#section-6.1.1